v0/destinations/google_adwords_remarketing_lists/transform.js

```diff
@@ -0,0 +1,272 @@
+  const userIdentifier = [];
+  const { typeOfList } = Config;
+  const { isHashRequired } = Config;
+  let attribute;
+  // setting attribute name according to type of list
+  switch (typeOfList) {
+    case "userID":
+      attribute = "thirdPartyUserId";
+      break;
+    case "mobileDeviceID":
+      attribute = "mobileId";
+      break;
+    default:
+      attribute = Config.userSchema;
+  }
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach((element, index) => {
+      if (isHashRequired) {
+        hashEncrypt(element);
+      }
+      // checking if the attribute is an array or not for generic type list
+      if (!Array.isArray(attribute)) {
+        if (element[`${attribute}`]) {
+          userIdentifier.push({ [attribute]: element[`${attribute}`] });
+        } else {
+          logger.log(` ${attribute} is not present in index:`, index);
+        }
+      } else {
+        attribute.forEach((attributeElement, index2) => {
+          if (attributeElement === "addressInfo") {
+            const addressInfo = constructPayload(element, addressInfoMapping);
+            // checking if addressInfo object is empty or not.
+            if (isDefinedAndNotNullAndNotEmpty(addressInfo))
+              userIdentifier.push({ addressInfo });
+          } else if (element[`${attributeElement}`]) {
+            userIdentifier.push({
+              [`${attributeMapping[attributeElement]}`]: element[
+                `${attributeElement}`
+              ]
+            });
+          } else {
+            logger.log(` ${attribute[index2]} is not present in index:`, index);
+          }
+        });
+      }
+    });
+  }
+  if (userIdentifier.length === 0)
+    throw new CustomError(
+      `[Google_adwords_remarketing_list]:: Their is not ${attribute} to put in userIdentifier.`,
+      400
+    );
+  return userIdentifier;
+};
+/**
+ * This function helps to create different operations by breaking the
+ * userIdentiFier Array in chunks of 20.
+ * Logics: Here for add/remove type lists, we are creating create/remove operations by
+ * breaking the userIdentiFier array in chunks of 20 and putting them inside one
+ * create/remove object each chunk.
+ * @param {rudder event message} message
+ * @param {rudder event destination} destination
+ * @returns
+ */
+
+const createPayload = (message, destination) => {
+  const { listData } = message.properties;
+
+  if (!listData.add || !isArray(listData.add)) {
+    throw new CustomError(
+      "[Google_adwords_remarketing_list]::add is not present inside listData. Aborting message.",
+      400
+    );
+  }
+  const userIdentidtifiersList = populateIdentifiers(listData.add, destination);
+  const outputPayload = constructPayload(message, offlineDataJobsMapping);
+  outputPayload.operations = [];
+  // breaking the userIdentiFier array in chunks of 20
+  const userIdentifierChunks = returnArrayOfSubarrays(
```

Comment:
For every offline data, jobs is the max limit for no of `userIdentifiers` is 20?
So for 100 user data it will look something like this?

`{
    "enablePartialFailure": true,
    "operations": [
      { 
         "create": {
            "userIdentifiers": [
              { "hashedEmail": "b4133380ccd33e9fea010c5c961c1b766f4bf70ca30a4026a65c6854c8921fda"},
              { "hashedEmail": "bc8ff7bc9cbcf078d9e537b9b6098f8d50fe7142086c6aabf6841a7714679c30"},
              { "hashedEmail": "4c7d03d328bed7e9babb9d5abb9216a1b14d5feeb82890816b090521072a5827"},
              { "hashedEmail": "d2b50b25bd11b7bb89b77196400c2224b7207e8a7311bb815eee5f4fdc6e84bd"},
              { "hashedEmail": "5a93546de6d673b505406bb492b1da7a05812cba88627fff7fe403f4504be0b5"},
              { "hashedEmail": "9740b36c2f55f133c73a11133b19db466b2f632aca80d3aef2cb00022f018f15"} ]
              .... X 20
          }
      },
       {
        ...
       } X 5
      ]
}`

Reply:
My question is what will happen when we have 100 userInfo in a single payload